### ISSUE TITLE
Is this mode collapse?

### ISSUE THREAD

**Issue by aptsurfer** [lucidrains/stylegan2-pytorch#183]
First of all, thank you lucidrains for being so active with this project. It is so cool!!

I am not myself much of a researcher, so I lack some intuition and understanding of some problems I have in training a dataset, which contains portraits of people. My dataset contains 50,000 images.

Training takes a long time on my GPU, about 20 minutes per 1k iterations, so I am trying to figure out if there is a problem early so I do not waste my resources. Here are the pictures I am getting after iteration 7,000 with default options:

![7](https://user-images.githubusercontent.com/69405481/101435883-09627800-3905-11eb-9709-da842ec5e2d6.jpg)

![7-ema](https://user-images.githubusercontent.com/69405481/101435899-0d8e9580-3905-11eb-8d25-7013f1622ac6.jpg)

![7-mr](https://user-images.githubusercontent.com/69405481/101435913-12534980-3905-11eb-8155-47211db8347e.jpg)

Is this an example of mode collapse? When all the pictures are the same, should I stop training and try something else?

I also tried it on a version of my dataset that had duplicate pictures (i.e. same person and pose) but with different backgrounds. Here is what it looked like at iteration 29k:

![29](https://user-images.githubusercontent.com/69405481/101436127-74ac4a00-3905-11eb-8346-581abb3f5f57.jpg)

![29-ema](https://user-images.githubusercontent.com/69405481/101436141-7970fe00-3905-11eb-8288-dd863a7db120.jpg)

![29-mr](https://user-images.githubusercontent.com/69405481/101436147-7c6bee80-3905-11eb-9cca-b0766889260c.jpg)

I also tried it once with these arguments, but after 6k iterations, I was just getting color gradients that were similar to each other (though not the same) so I stopped it:

`stylegan2_pytorch --data=./portraits --dataset-aug-prob 0.05 --trunc-psi 0.8 --cl-reg`

Any tips? Thank you so much!

**Comment by lucidrains** [comment]
That is indeed mode collapse! You should try increasing the gradient accumulation steps '--gradient-accumulate-every'

50k images is usually enough to avoid mode collapse, so I'm suspicious whether your images are in the right format to be picked up by the script. The training script only accepts images of png and jpeg format, just to put out a caution

**Comment by lucidrains** [comment]
Also try putting in a slight augmentation with --aug-prob of 0.1

**Comment by lucidrains** [comment]
@aptsurfer if you are willing to wait a bit, i'll try to add the autoencoding self-supervised discriminator loss from 'Lightweight' GAN and see if that helps with mode collapse and training stability. prob by EOW

**Comment by aptsurfer** [comment]
First of all, thank you for being so responsive and volunteering your time - that makes me smile! I did verify that the program found all my images (a print statement I added to the Dataset class said it found all 50k) - so that's good. Then I used the arguments you suggested (--gradient-accumulate-every=16 --aug-prob=0.1). In the first 4k iterations, the output was mostly the same type (this is at 4k):
![4](https://user-images.githubusercontent.com/69405481/101567273-6bd07c80-39c8-11eb-83f0-5345233eddfd.jpg)

But soon after, it picked up more and more variations. I ran it for the past 23 or so hours and am beginning to get some very good results!

I noticed that the generator/discriminator score has been flipping a lot more. So I guess that competition helped improve learning. In the first 4k iterations (and in my previous attempts) the generator score seemed to be higher the whole time.

Here's a copy of my log so far for anyone who's interested:
[portraits_output_24k.txt](https://github.com/lucidrains/stylegan2-pytorch/files/5663242/portraits_output_24k.txt)
Maybe it can explain something about its progress.

Training does take longer than before. Instead of 20 minutes per 1k iterations, it is more like 55 minutes. But the quality after only one day is looking good, so perhaps that is not as relevant.
